### PR TITLE
Ensure time picker uses correct color/typography values

### PR DESCRIPTION
**Pull request by rami-a**
This PR updates color/typography to ensure defaults adhere to the spec.

### CODE REVIEW

File: packages/flutter/lib/src/material/time_picker.dart [hunk 1]
```diff
@@ -1259,6 +1250,7 @@ class _DialState extends State<_Dial> with SingleTickerProviderStateMixin {
           secondaryLabels: secondaryLabels,
           backgroundColor: backgroundColor,
           accentColor: accentColor,
+          dotColor: Theme.of(context).colorScheme.surface,
```
Comment (clocksmith) [inline packages/flutter/lib/src/material/time_picker.dart:1253]: use already defined `theme`

File: packages/flutter/lib/src/material/time_picker.dart [hunk 2]
```diff
@@ -1209,18 +1211,7 @@ class _DialState extends State<_Dial> with SingleTickerProviderStateMixin {
...
-          break;
-        case Brightness.dark:
-          backgroundColor = themeData.colorScheme.surface;
-          break;
-      }
-    }
-
+    final Color backgroundColor = TimePickerTheme.of(context).dialBackgroundColor ?? themeData.colorScheme.onBackground.withOpacity(0.12);
     final Color accentColor = TimePickerTheme.of(context).dialHandColor ?? themeData.colorScheme.primary;
     final ThemeData theme = Theme.of(context);
```
Comment (clocksmith) [inline packages/flutter/lib/src/material/time_picker.dart:1216]: create `pickerTheme`?

File: packages/flutter/test/material/time_picker_theme_test.dart [hunk 3]
```diff
@@ -233,19 +224,18 @@ void main() {
...
     dialBackgroundColor: Colors.pinkAccent,
-    activeDayPeriodColor: Colors.teal,
-    hourMinuteTextStyle: const TextStyle(fontSize: 8),
-    dayPeriodTextStyle: const TextStyle(fontSize: 8),
-    helpTextStyle: const TextStyle(fontSize: 8),
-    shape: const RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(16.0))),
-    hourMinuteShape: const RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(16.0))),
-    dayPeriodShape: const RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(16.0))),
-    inputDecorationTheme: const InputDecorationTheme(
+    hourMinuteTextStyle: TextStyle(fontSize: 8),
```
Comment (clocksmith) [inline packages/flutter/test/material/time_picker_theme_test.dart:232]: nit 8 vs 8.0 and 16 vs 16.0

File: packages/flutter/lib/src/material/time_picker.dart (hunk 2 above)
Comment (rami-a) [reply packages/flutter/lib/src/material/time_picker.dart:1214]: done

File: packages/flutter/test/material/time_picker_theme_test.dart (hunk 3 above)
Comment (rami-a) [reply packages/flutter/test/material/time_picker_theme_test.dart]: done

File: packages/flutter/lib/src/material/time_picker.dart (hunk 1 above)
Comment (rami-a) [reply packages/flutter/lib/src/material/time_picker.dart]: done

File: packages/flutter/lib/src/material/time_picker.dart [hunk 4]
```diff
@@ -484,8 +484,8 @@ class _DayPeriodControl extends StatelessWidget {
   Widget build(BuildContext context) {
     final MaterialLocalizations materialLocalizations = MaterialLocalizations.of(context);
     final ColorScheme colorScheme = Theme.of(context).colorScheme;
-    final Color activeBackgroundColor = TimePickerTheme.of(context).activeDayPeriodColor ?? colorScheme.surface;
-    final Color backgroundColor = colorScheme.onBackground.withOpacity(0.06);
+    const Color activeBackgroundColor = Colors.transparent; // So that it matches the rest of the dialog.
```
Comment (willlarche) [inline packages/flutter/lib/src/material/time_picker.dart:487]: Clarify if possible.

Comment (rami-a) [reply packages/flutter/lib/src/material/time_picker.dart]: Done

**Pull request merged**